server/gfmd/user.c

```diff
@@ -419,6 +355,154 @@ user_enter_auth_id(
 	return (GFARM_ERR_NO_ERROR);
 }
 
+static gfarm_error_t
+user_auth_id_modify_internal(struct user *u,
+		char *auth_id_type, char *auth_user_id,
+		bool *need_to_update_dbp, bool *need_to_addp)
+{
+	enum auth_user_id_type auth_user_id_type = 0;
+	char *old_auth_user_id = NULL;
+	struct gfarm_hash_entry *entry;
+	gfarm_error_t e;
+
+	*need_to_update_dbp = false;
```

Comment:
This is OK for this code, but in general it is better to set this kind of return value only for GFARM_ERR_NO_ERROR.
Think about the following code, the return value of some_other_function() will be lost.
```
e = some_other_function(, ... &need_to_update_db);
if (e == GFARM_ERR_NO_ERROR) {
  e2 = user_auth_id_internal(, ... &need_to_update_db);
  /* an error of e2 is not fatal */
}
if (need_to_update_db) ...
```